client/src/LayoutComponents/AllStudents/AllStudents.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import searchIcon from '../../images/search.svg';
+import axios from 'axios';
+import StudentCard from '../../SharedComponents/StudentCard';
+import './AllStudents.css';
+import { Link } from 'react-router-dom';
+
+class AllStudents extends Component {
+  state = {
+    searchfield: '',
+    students: [],
+  };
+
+  componentDidMount() {
+    axios
+      .get(`/api/students`)
+      .then((res) => {
+        this.setState({
+          students: res.data,
+        });
+      })
+      .catch((err) => err);
+  }
+
+  // store the input of the search field
+  onSearchChange = (event) => {
+    this.setState({ searchfield: event.target.value });
```

Comment:
It might be helpful to trim() the value before storing it, so any leading / trailing whitespace that the user types is ignored when searching.